deepfence_server/reporters/scan/scan_result_actions.go

```diff
@@ -257,37 +258,35 @@ func StopCloudComplianceScan(ctx context.Context, scanType, scanId string) error
 }
 
 func StopScan(ctx context.Context, scanType, scanId string) error {
-
 	driver, err := directory.Neo4jClient(ctx)
 	if err != nil {
 		return err
 	}
 	session := driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeWrite})
 	defer session.Close()
+	txConfig := neo4j.WithTxTimeout(15 * time.Second)
 
-	tx, err := session.BeginTransaction(neo4j.WithTxTimeout(15 * time.Second))
-	if err != nil {
-		return err
-	}
-	defer tx.Close()
+	nodeStatusField := ingestersUtil.ScanStatusField[utils.Neo4jScanType(scanType)]
+	latestScanIdField := ingestersUtil.LatestScanIdField[utils.Neo4jScanType(scanType)]
 
-	query := `MATCH (n:%s) -[:SCANNED]-> ()
+	query := `MATCH (n:%s) -[:SCANNED]-> (m)
         WHERE n.node_id = $scan_id
         AND n.status IN [$starting, $in_progress]
-        SET n.status=$cancel_pending`
-
-	if _, err = tx.Run(fmt.Sprintf(query, scanType),
+		WITH n,m
+        SET n.status= CASE WHEN n.status = $starting THEN $cancelled 
+		ELSE $cancel_pending END, 
+		m.%s=n.status, m.%s=$scan_id`
+	_, err = session.Run(fmt.Sprintf(query, scanType, nodeStatusField, latestScanIdField),
```

Comment:
We should share the transaction between all the stop Scan & do 1 final commit at the end.
The reasoning being, if we request N stops and one fails, all N should fail, otherwise it's hard to keep track of what was cancelled and not cancelled.
The easiest and cleanest thing to do is to tight them up, the same way we do in start.

Reply:
Done.